### PR TITLE
Added gradient function to JAX frontend

### LINKED ISSUE #10838: gradient
- [ ] #8426

### PR DESCRIPTION
**Pull request by Supremolink81**
Close #10838

jax.numpy.gradient documentation: .readthedocs.io/en/latest/_autosummary/jax.numpy.gradient.html#jax.numpy.gradient

### CODE REVIEW

**Comment by juliagsy** [comment]
Hey there!

We rebased the repository recently which caused the 10000 commits, please kindly update your fork and copy over your changes in the related files only a new PR. After that, you may close this PR!

We sincerely apologise for any inconvenience caused, and thanks a lot for your understanding and cooperation!
